dxr/build.py

```diff
@@ -539,8 +538,26 @@ def index_file(tree, tree_indexers, path, es, index, jinja_env):
 
              'is_text': is_text,
 
+             # if the file's an image, use <img src=image_src>
+             'image_src': image_src,
+
              'sections': build_sections(links)})
 
+def index_image(tree, rel_path, index, es):
+    '''
+    Index the image located at path relative to given tree using es in index.
+    Return the URL to get the image.
+    '''
+    # define the image source
+    image_src = "%s/images/%s" % (tree.name, rel_path)
+    # encode the image file as base64 string and index it
+    with open(join(tree.source_folder, rel_path), "rb") as img:
+        imgdata = img.read().encode("base64")
+        es.index(index, IMAGE, {
```

Comment:
You've got the right idea here, but I think it would be more consistent with the existing ES schema if we put the image in a field of FILE rather than introducing a new IMAGE doctype. We'd then tweak app.browse() to pull only the fields it needs when building dir listings (so it doesn't pull down huge images all the time). This would also simplify a future feature: thumbnails. It would become trivial to store a thumbnail-sized image in a FILE field as well, and that could be fetched in the same request as the rest of the file data in browse().